src/utils/utils.ts

```diff
@@ -0,0 +1,64 @@
+import { ExceptionType } from '../constant';
+import config from '../config';
+import { AppActingUser, AppCallResponse, Oauth2App } from '../types';
+
+import {
+    newErrorCallResponseWithMessage,
+    newOKCallResponseWithMarkdown,
+} from './call-responses';
+import { Exception } from './exception';
+
+export function isUserSystemAdmin(actingUser: AppActingUser): boolean {
+    return Boolean(actingUser.roles && actingUser.roles.includes('system_admin'));
+}
+
+export function existsOauth2App(oauth2App: Oauth2App): boolean {
+    return Boolean(oauth2App.client_id) && Boolean(oauth2App.client_secret);
+}
+
+export function existsToken(oauth2App: Oauth2App): boolean {
+    const oauthUser = oauth2App.user;
+    return Boolean(oauthUser?.token);
+}
+
+export function replace(value: string, searchValue: string, replaceValue: string): string {
+    return value.replace(searchValue, replaceValue);
+}
+
+export function errorDataMessage(error: Exception | Error | any): string {
+    const errorMessage: string = error?.response?.data || error?.response?.data?.message || error?.message || error;
+    return `${errorMessage}`;
+}
+
+export function tryPromise(p: Promise<any>, exceptionType: ExceptionType, message: string) {
```

Comment:
This would make `tryPromise` more useful:

```suggestion
export function tryPromise<T>(p: Promise<T>, exceptionType: ExceptionType, message: string): Promise<T> {
```